core/domain/suggestion_jobs_one_off.py

```diff
@@ -0,0 +1,108 @@
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import ast
+import logging
+
+from core import jobs
+from core.domain import html_validation_service
+from core.domain import suggestion_registry
+from core.domain import suggestion_services
+from core.platform import models
+
+(suggestion_models,) = models.Registry.import_models([models.NAMES.suggestion])
+
+
+class SuggestionMathRteAuditOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that checks for existence of math components in the suggestions."""
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [suggestion_models.GeneralSuggestionModel]
+
+    @staticmethod
+    def map(item):
+        html_string_list = []
+        suggestion = suggestion_services.get_suggestion_from_model(item)
+        html_string_list = (
+            html_string_list + suggestion.get_all_html_content_strings())
+        html_string = ''.join(html_string_list)
+        if html_validation_service.check_for_math_component_in_html(
+                html_string):
+            yield ('Suggestion with Math', item.id)
+
+    @staticmethod
+    def reduce(key, values):
+        yield (
+            '%d suggestions have Math components in them with IDs.' %
+            len(values), values)
+
+
+class SuggestionMathMigrationOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """A one-time job that can be used to migrate the Math components in the
+    suggestions to the new Math Schema.
+    """
+    _ERROR_KEY = 'validation_error'
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [suggestion_models.GeneralSuggestionModel]
+
+    @staticmethod
+    def map(item):
+        suggestion = suggestion_services.get_suggestion_by_id(item.id)
+        try:
+            suggestion.validate()
+        except Exception as e:
+            logging.error(
+                'Suggestion %s failed validation: %s' % (item.id, e))
+            yield (
+                SuggestionMathMigrationOneOffJob._ERROR_KEY,
+                'Suggestion %s failed validation: %s' % (item.id, e))
+            return
+        suggestion_domain_class = (
+            suggestion_registry.SUGGESTION_TYPES_TO_DOMAIN_CLASSES[
+                suggestion.suggestion_type])
+        suggestion.change = (
+            suggestion_domain_class.convert_html_in_suggestion(
+                suggestion.change.to_dict(),
+                html_validation_service.
+                add_math_content_to_math_rte_components))
+        try:
+            suggestion.validate()
+        except Exception as e:
+            logging.error(
+                'Suggestion %s failed after validation: %s' % (item.id, e))
+            yield (
+                SuggestionMathMigrationOneOffJob._ERROR_KEY,
+                'Suggestion %s failed after validation: %s' % (item.id, e))
+            return
+        item.change_cmd = suggestion.change.to_dict()
+        item.put()
```

Comment:
Will last_updated change during this put()? Should it? (Add tests accordingly.)

Reply:
Yeah, the last_updated is changed during the put.
I think it should update, because other migrations also update this field.
Added tests

Reply:
I don't think it should update the last updated field... You can perform the migration without affecting the last updated field I think...

Reply:
Just confirming, last_updated will not change during this right? That isn't desired specifically for suggestion/feedback models, because of 2 things... Firstly, we display the timestamp in the UI based on the last updated field and updating it in the job would mess up things. We order the suggestions based on the last updated field and surface the most recent ones higher up. If last updated is overwritten, this would get messed up too...